File: OCH_Tao/ABC/B/128.py
N = int(input())
X = []
L = set()
for i in range(N):
  S,P = input().split()
  L.add(S)
  X.append([S,int(P),i+1])
L = sorted(list(L))
for i in L:
  l = [j for j in X if j[0]==i]
  l = sorted(l,key=lambda x:x[1],reverse=True)
  for k in l:
    print(k[2])

File: OCH_Tao/ABC/B/128.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input.shift(), 10);
const L = [];
let Y = new Set();
let i = 1;
for (const X of input) {
  const [S, P] = X.split(" ");
  Y.add(S);
  L.push([S, parseInt(P, 10), i]);
  i++;
}
let Z = [...Y];
Z.sort();
for (const X of Z) {
  let l = L.filter(x => x[0] === X);
  l.sort((a, b) => b[1] - a[1]);
  for (const i of l) {
    console.log(i[2]);
  }
}